### PR TITLE
[21.11] imagemagick: apply upstream patch to fix perlPackages.ImageMagick

### PR DESCRIPTION
**Pull request by wamserma**
###### Motivation for this change

https://github.com/NixOS/nixpkgs/issues/156826
Backport of https://github.com/NixOS/nixpkgs/pull/156959

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by wamserma** [comment]
@wentasah feel free to cherry-pick this on unstable, I made this fix first on release to avoid downloading too many build dependencies

**Comment by wamserma** [comment]
@ofborg build perlPackages.ImageMagick

File: pkgs/applications/graphics/ImageMagick/7.0.nix [hunk 1]
```diff
@@ -27,6 +27,8 @@ stdenv.mkDerivation rec {
     sha256 = "0r8zmk2cfmf09l94hqzfz4aspnzn178ggdbgm7w4hr0p864cbvc3";
   };
 
+  patches = [ ./revert-075565e93c71bcaaabf0ce70b7d1060bccdf0020.patch ];
```
Comment (cole-h) [inline pkgs/applications/graphics/ImageMagick/7.0.nix:30]: Was there an issue filed upstream about this? If so, could you please add a link to it in a comment above this patch?

**Comment by erictapen** [comment]
Thanks for the PR. I saw that upstream seems to have fixed the issue in https://github.com/ImageMagick/ImageMagick/commit/62845d5672eca4446b952dd0ab2e3e0dab0309d4. Could you fetch that commit instead (via fetchpatch) and target `master`? We'll backport to 21.11 then.

File: pkgs/applications/graphics/ImageMagick/7.0.nix (hunk 1 above)
Comment (wamserma) [reply pkgs/applications/graphics/ImageMagick/7.0.nix:30]: Yes, and upstream provided a fix within minutes. Will update the PR with the upstream patch.

**Comment by wamserma** [comment]
> Could you fetch that commit instead (via fetchpatch) and target `master`?

Done. Made this PR a backport.

**Comment by dasJ** [comment]
Since the master PR is merged and this is now a cherry-pick of the master commit, this is probably good to go

**Comment by erictapen** [comment]
For reference: That PR to master was #156959

**Comment by bjornfor** [comment]
Please create backport PRs from the master branch instead of directly from unmerged PRs. This PR was now merged when pointing to a nonexisting branch. (Applying a backport label to the master PR automatically creates such a backport PR with correct ref.)

**Pull request merged**